Display appropriate messages instead of empty tables when there are no entries in the database

kushal-dreamspring opened the pull request:
- Display the message "No invoice found" when there are no invoices present
- Display the message "No cars parked" when there are no cars in the parking lot

kushal-dreamspring committed:
Display appropriate messages instead of empty tables when there are no entries in the database
- Display the message "No invoice found" when there are no invoices present
- Display the message "No cars parked" when there are no cars in the parking lot

jinifrancis commented on app.rb:
```diff
@@ -78,17 +78,26 @@ def find_and_unpark_car(registration_number)
-    puts "Invoice number\tRegistration Number\tEntry Time\t\t\tExit Time\t\t\tDuration\tAmount"
-    invoices.each do |invoice|
-      puts "#{invoice[:id]}\t\t#{invoice[:registration_number]}\t\t#{invoice[:entry_time]}\t#{invoice[:exit_time]}\t#{invoice[:duration]}\t\t#{invoice[:invoice_amount]}"
+    if !invoices.empty?
```
We can do away with the negation here since the condition has actions for `if..else`, ie,
```
if invoices.empty?  
    p 'No invoice'. 
else 
    p 'invoices'
end
```

jinifrancis commented on app.rb:
```diff
@@ -78,17 +78,26 @@ def find_and_unpark_car(registration_number)
+        puts "#{car[:id]}\t#{car[:registration_number]}\t\t#{car[:entry_time]}"
+      end
+    else
+      puts 'No Car Found'
```
Same point as before. We can do away with the negation before empty? check

jinifrancis commented on app.rb:
I don't see corresponding spec updates for the changes you made here. Add specs to check these actions.

jinifrancis replied:
Remember to mark feedback items as done once you complete the changes required for each comment.

kushal-dreamspring committed:
Remove negation from no cars and no invoice check and add specs for these

- Add Specs for no invoice and no parked cars
- Add multiple contexts in acceptance tests
- Remove negation from no invoice check in print_all_invoice action
- Remove negation from no car check in print_all_parked_cars action

jinifrancis commented on spec/acceptance/parking_lot_spec.rb:
```diff
@@ -35,40 +59,38 @@
+    it 'list all the invoices' do
+      expect { system('RACK_ENV="test" ./app.rb --all-invoices') }
+        .to output(
+              /Invoice number\tRegistration Number\tEntry Time\t\t\tExit Time\t\t\tDuration\tAmount
```
Remove tabs and use spaces

jinifrancis commented on spec/acceptance/parking_lot_spec.rb:
```diff
@@ -15,16 +15,40 @@
+    it 'should print \'No Invoice Found\'' do
+      expect { system('RACK_ENV="test" ./app.rb --all-invoices') }
+        .to output("No Invoice Found\n").to_stdout_from_any_process
+    end
```
Here I see that the specs are for 3 separate actions - park car, print all cars and print all invoices. Add additional text or contexts for better readability of the specs

kushal-dreamspring committed:
Fix Acceptance Specs

- Add action context to acceptance specs
- Add parking acceptance spec for when parking lot is full
- Add invoice acceptance spec for when no invoice is present
- Add all parked cars acceptance spec for when when no car is parked

kushal-dreamspring committed:
Resolve Merge Conflict with fix/unpark_message

kushal-dreamspring committed:
Resolve Conflict with merging with fix/unpark_message

jinifrancis reviewed (approved):
You can merge this PR
